scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/LiveJobs.java

```diff
@@ -614,6 +614,81 @@ private void setJobStatusToInErrorIfNotPaused(InternalJob job) {
         }
     }
 
+    TerminationData finishInErrorTask(JobId jobId, String taskName) throws UnknownTaskException, UnknownJobException {
+        JobData jobData = lockJob(jobId);
+        if (jobData == null) {
+            throw new UnknownJobException(jobId);
+        }
+        InternalJob job = jobData.job;
+        try {
+            InternalTask task = job.getTask(taskName);
+            if (task == null) {
+                throw new UnknownTaskException(taskName);
+            }
+            
+            TaskId taskId = task.getId();
+            if (!task.getStatus().isTaskAlive()) {
+                tlogger.info(task.getId(), "task isn't alive: " + task.getStatus());
+                return emptyResult(task.getId());
+            }
+
+            TaskResultImpl taskResult = new TaskResultCreator()
+                    .getEmptyTaskResultWithTaskIdAndExecutionTime(task);
+            
+            RunningTaskData data = new RunningTaskData(task, job.getOwner(), 
+                    job.getCredentials(), task.getExecuterInformation().getLauncher());
+            
+            TerminationData terminationData = TerminationData.newTerminationData();
+            terminationData.addTaskData(job, data, false, taskResult);
+
+            if (onErrorPolicyInterpreter.requiresCancelJobOnError(task)) {
```

Comment:
I don't think we need to check if the task requires Cancel Job On Error. Because the task actually failed already, so the Cancel Job On Error was already triggered when the task failed. 
I think you can remove that if.